File: MonitoreoCNET/MonitoreoCNET/Models/Servicios/PrestacionDeServicio.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MonitoreoCNET.Models.Organizaciones;
using MonitoreoCNET.Models.Servicios;

namespace MonitoreoCNET.Models.Servicios
{
    [Table("prestacion_de_servicio")]
    public class PrestacionDeServicio
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [ForeignKey("id_servicio")]
        public virtual Servicio Servicio { get; set; }

        [ForeignKey("id_establecimiento")]
        public virtual Establecimiento Establecimiento { get; set; }

        public PrestacionDeServicio(Servicio servicio, Establecimiento establecimiento)
        {
            Servicio = servicio;
            Establecimiento = establecimiento;
        }

        public PrestacionDeServicio()
        {
        }
    }
}

File: monitoreo/src/main/java/models/entidades/servicios/servicio/PrestacionDeServicio.java
package models.entidades.servicios.servicio;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import models.entidades.organizaciones.Establecimiento;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "prestacion_de_servicio")
public class PrestacionDeServicio {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Setter @Getter
  private int id;

  @Getter @Setter
  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "id_servicio", referencedColumnName = "id")
  private Servicio servicio;

  @Getter @Setter
  @JsonManagedReference
  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "id_establecimiento", referencedColumnName = "id")
  private Establecimiento establecimiento;

  public PrestacionDeServicio(Servicio servicio, Establecimiento establecimiento){
    this.servicio = servicio;
    this.establecimiento = establecimiento;
  }

  public PrestacionDeServicio() {
  }

}
